### ISSUE TITLE
Make assignment 1

### ISSUE THREAD

**Issue by rrheddema** [rrheddema/CS231n#2]
**Actionplan**
- get started with http://cs231n.github.io/assignments2017/assignment1/
- check in solution, use also year in folder structure to enable checkin of other years assignments too.
- cross check solution with https://github.com/MahanFathi/CS231

- [x] Create setup
- [x] Q1: k-Nearest Neighbor classifier
- [x] Q2: Training a Support Vector Machine
- [x] Q3: Implement a Softmax classifier
- [x] Q4: Two-Layer Neural Network
- [x] Q5: Higher Level Representations: Image Features

**Schedule**
- Start: 31 jul 2018
- Finish: 14 aug 2018
- let's define a milestone 18CW33 for that.

**Comment by rrheddema** [comment]
**Done yesterday:** 

Experimenting with:
- Create a docker file to create a container as work environment to keep own system clean
- With volume mapping to access the assignments files/dataset: **OK**
- With port forwarding to enable using jupyter: **In progress**
- Documenting it with a markdown file

**Plan for today.**
- Checkin of Dockerfile and markdown file.

**Comment by rrheddema** [comment]
**Done**
Dockerfile added and documentation updated **now!**

**Plan for today**
Getting starting with the k-Nearest Neighbor classifier.

**Comment by rrheddema** [comment]
**Done**
Encountered some errors when attempting to run the notebook and starting its steps.
Resolved issues by installing additional packages/libraries
Adapted dockerfile and readme.md accordingly  and committed to the repos.
Implemented the compute_distances_two_loops, finished 8 steps of 16. At 50% now, with knn.

**Plan for today**
Continue with step 9-12, hope to complete it to 75% regarding Q1

**Progress with respect to the target**
17 august was targeted, looking good! :smile:

**Comment by rrheddema** [comment]
**Done**
Finished k-Nearest Neighbor classifier in the Notebook.

**Plan for today**
- Document work done/remarks in a markdown document
- Commit it.

**Progress with respect to the target**
Finishing Q1 took 3 weeks (1 week holiday).
Projecting it to finish the other 4 assignments, +8 weeks from now.  :scream:

New target to finish it: 12 oct 2018  :running:

**Comment by rrheddema** [comment]
Commited https://github.com/rrheddema/CS231n/commit/1f65bd0085e17b1ae2b72515343242b00a57f95c

**Comment by rrheddema** [comment]
**Done**
Finished SVM in the Notebook.

**Plan for today**
- Document work done/remarks in a markdown document
- Commit it.

**Comment by rrheddema** [comment]
**Done**
Finished Softmax in the Notebook.

**Plan for today**
- Document work done/remarks in a markdown document
- Commit it.

**Progress with respect to the target**
- Spent on Q3, 1 week, projecting this on 2 remaining tasks: 20 september.
- Previous projecting was 12 october.
- Let's target something in between....

New target to finish it: 4 Oct 2018

**Comment by rrheddema** [comment]
Commited softmax with 70be94e06ba089123b2c356a165d09a2aecdabf2

**Comment by rrheddema** [comment]
**Done**
- Finished two layer net in the notebook.
- Committed results https://github.com/rrheddema/CS231n/commit/26429caf49e8b4eeaec344120f944d4949809b2c

**Plan for today**
- Start with "High level representations: Image Features"

**Comment by rrheddema** [comment]
**Done**
- Commited results https://github.com/rrheddema/CS231n/commit/f41ac2e13543e8d25a436e5f6c33df0573c5f21a

**Closing this issue**
